### DISCUSSION TITLE
Clarification on the usage of nodemix and nodeofactor/nodeifactor

### DISCUSSION THREAD

**Discussion by gocoyd** [statnet/ergm#557] [Q&A]
Hello!

I am very new to ERGM and have some questions about model specification. I am interested in ethnic homophily in school classrooms and to get the measures I use nodemix('minor', base = c(2, 3)). As a result it provides me with migrant homophily (mix.minor.1.1) and native homophily (mix.minor.0.0) rates. I do not use just nodemix because I want to get more detailed results.
At the same time I want to account for sender and receiver effects, so I use nodeofactor('minor') and nodeifactor('minor') but the model gives a warning:

`Model statistics ‘mix.minor.1.1’ are linear combinations of some set of preceding statistics at the current stage of the estimation. This may indicate that the model is nonidentifiable.`

The model works okay when I use either nodeofactor('minor') or nodeifactor('minor') and not them together. Or if I use nodematch instead of nodemix. Can somebody please explain why this is so? I would be very grateful.

**Comment by krivit** [comment]
Firstly, please take a look at the term help, i.e., `ergmTerm?nodemix`, `ergmTerm?nodeofactor`, etc. if you haven't already. In particular, using `base=` is deprecated.

Secondly, in the future, please provide more complete examples, i.e., the formulas for the models you are trying to fit.

Based on what you've written, if you have an directed network and a nodal attribute with two levels, counting up the ties between and within them gives you four distinct counts: 0-0, 0-1, 1-0, and 1-1. This means that you can have at most a total of 4 statistics involving them, and how you parametrise them depends on your substantive considerations. In this case, I suspect your model also has an edges effect, so you have nodeofactor.minor.1 + nodeifactor.minor.1 - mix.minor.1.1 + mix.minor.0.0 = edges (for every possible network), so one of these is redundant, which is what that message is telling you.

If this doesn't answer your question, please reopen the discussion.
